common-ktx/src/androidTest/java/co/nimblehq/common/extensions/SharedPreferencesExtTest.kt

```diff
@@ -0,0 +1,121 @@
+package co.nimblehq.common.extensions
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.SmallTest
+import androidx.test.platform.app.InstrumentationRegistry
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.isA
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@SmallTest
+class SharedPreferencesExtTest {
+    private var context: Context = InstrumentationRegistry.getInstrumentation().context
```

Comment:
Hmmm. I am still thinking about using Android Instrumentested Test for this extension. I think we could use UnitTest instead. How do you think?

Reply:
@hoangnguyen92dn Doing Unit test with Mock is not a good idea IMO as it can not prove our concept 🙏

Reply:
@doannimble Can't we just assert before and after we save the value on the sharePreferences? Just curious.

Reply:
@lydiasama I think we will do if we need to do some stuff things before and after the testing. In this case, it is just a simple assignment of variable, hence, assert before and after look cumbersome 🙏

Reply:
@doannimble The Instrumentested Test requires an Android environment to execute the test case. That's why I suggest using UnitTest instead. Btw, please update the test case name to follow our convention `When...` as well? 😄